src/librustc_resolve/lib.rs

```diff
@@ -3229,9 +3226,12 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             }
         }
 
+        // As a loose rule to avoid obviously incorrect suggestions, clamp the
+        // maximum edit distance we will accept for a suggestion to one third of
+        // the typo'd name's length.
+        let max_distance = std::cmp::max(name.len(), 3) / 3;
```

Comment:
Because we're dividing the `max(name.len(),3)` by `3`, isn't the `max_distance` always 1 now? It seems like we'd want this to be `let max_distance = std::cmp::max(name.len() / 3, 3);`, instead.